### PR TITLE
Support conan.cmake in exported packages

### PR DESCRIPTION
**Pull request by sixten-hilborn**
See #10.

### CODE REVIEW

File: conan.cmake [hunk 1]
```diff
@@ -197,19 +197,30 @@ endmacro()
 
 macro(conan_cmake_run)
     parse_arguments(${ARGV})
-    conan_cmake_generate_conanfile(${ARGV})
-    if(CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
+
+    # TODO: Better detection
+    if(CONAN_EXPORTED OR CONAN_COMPILER)
```
Comment (memsharded) [inline conan.cmake]: I am not sure if I'd use CONAN_COMPILER for detection, some users might be feeding it from the command line to cmake. CONAN_EXPORTED would do the task, wouldn't it? Is it necessary to set it to ``ON`` then?

File: tests.py [hunk 2]
```diff
@@ -169,3 +169,42 @@ class Pkg(ConanFile):
       run("cmake .. -DCMAKE_BUILD_TYPE=Release")
       run("cmake --build . --config Release")
       run("bin\main")
+
+    def test_exported_package(self):
+      content = """cmake_minimum_required(VERSION 2.8)
+project(conan_wrapper CXX)
+
+set(CONAN_EXPORTED ON)
```
Comment (memsharded) [inline tests.py]: I guess we should remember to remove this ``CONAN__EXPORTED ON`` line, when the conan codebase already sets it automatically.

File: conan.cmake (hunk 1 above)
Comment (sixten-hilborn) [inline conan.cmake]: Agree, this is a temporary workaround before CONAN_EXPORTED was defined by conan, but it's supposed to work with both 0.18 and 0.19.

File: tests.py (hunk 2 above)
Comment (sixten-hilborn) [inline tests.py]: That's true, it's not needed from the next conan release.

**Pull request merged**